src/components/buttonSelectors/MultiButtonSelector.js

```diff
@@ -1,117 +1,159 @@
+  spreadsheet.
+ */
+import React from "react";
+import { connect } from "react-redux";
+import { View, Image } from "react-native";
+import { Box, Text, VStack, Button, HStack } from "native-base";
+import QuestionSection from "../QuestionSection";
+import ImageSelector from "../../image/imgIndex";
+import { updateResponse } from "../../actions/StoryActions";
+import TooltipView from "../Tooltip.js";
+import { dependencyParser } from "../../utils/dependencyHelper";
+import { SafeAreaView } from "react-navigation";
 
 const MultiButtonSelector = (props) => {
-    const [selection, setSelection] = React.useState(null);
-    const {data, key, id, questionReducer, submitFunction, updateResponse, dependencyID} = props;
+  const [selection, setSelection] = React.useState(null);
+  const {
+    data,
+    key,
+    id,
+    questionReducer,
+    submitFunction,
+    updateResponse,
+    dependencyID,
+  } = props;
 
-    let currId = data.humanReadableId
+  let currId = data.humanReadableId;
 
-    const reducerData = questionReducer.data.find(entry => entry.id == id);
-    let existingData = !reducerData?.response ? null: reducerData.response;
+  const reducerData = questionReducer.data.find((entry) => entry.id == id);
+  let existingData = !reducerData?.response ? null : reducerData.response;
 
-    // Populate if value already exists in redux
-    if(!selection) {
-        if(existingData != null) {
-            if(existingData[currId] != selection) {
-                setSelection(existingData[currId]);
-            }
-        }
+  // Populate if value already exists in redux
+  if (!selection) {
+    if (existingData != null) {
+      if (existingData[currId] != selection) {
+        setSelection(existingData[currId]);
+      }
     }
+  }
 
-
-    const submitField = (optionText, idCode) => {
-        setSelection(optionText);
-        if (dependencyID==null || dependencyID.length == 1){
-            updateResponse && updateResponse({id, question: currId, selection: idCode})
-        }else{
-            let vehicleID = dependencyID[1]
-            switch (dependencyID.length) {
-                case 2:
-                    updateResponse && updateResponse({id, question: currId, selection: idCode, vehicleID: vehicleID})
-                    break;
-                case 3:
-                    let passengerID = dependencyID[2]
-                    updateResponse && updateResponse({id, question: currId, selection: idCode, vehicleID: vehicleID, passengerID: passengerID})
-                case 4:
-                    updateResponse && updateResponse({id, question: currId, selection: idCode, vehicleID: vehicleID, nonmotoristID: dependencyID[3]})
-                default:
-                    break;
-            }
-        }
-        submitFunction({id, question: currId, selection: optionText})
+  /*
+    This function saves the value of the  selected button to the form reducer.
```

Comment:
elaborate that it checks which form the question is in and updates the appropriate reducer